app/src/main/java/ch/epfl/sdp/peakar/user/profile/ProfileLauncherActivity.java

```diff
@@ -37,45 +50,68 @@ protected void onCreate(Bundle savedInstanceState) {
         getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                 WindowManager.LayoutParams.FLAG_FULLSCREEN);
 
-        authService = AuthService.getInstance();
+        if(!Database.getInstance().isOnline()) {
+            AlertDialog.Builder builder = new AlertDialog.Builder(this);
+            builder.setMessage(ProfileOutcome.FAIL.getMessage())
+                    .setCancelable(false)
+                    .setPositiveButton("Ok", new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialog, int id) {
+                            finish();
+                        }
+                    });
```

Comment:
Can be replaced by lambda expression.
```suggestion
                    .setPositiveButton("Ok", (dialog, id) -> finish());
```